src/schema/entities.yaml

```
@@ -1,11 +1,13 @@
 ---
-sub:
+subject:
   name: Subject
+  entity: sub
```

> Could you make your suggestion concrete?

don't treat the key as the pybids variable.

```suggestion
subject:
  name: Subject
  entity: sub
  variable_name: subject
```

this results in a duplicate name but leaves subject the key to be an enumerated concept/term in bids.

---

Is there a case where this isn't duplicative? If the key will always be a valid variable name, then pybids might be able to move toward whatever is decided as the key. If not, then being explicit makes sense.

While this is phrased as a way to standardize variable names, the pybids variable names were chosen to solve the problem of referencing the concepts more clearly, although constrained by Python syntax.

---

as you get to other concepts like metadata you may run into variable names being different or shorter again. we may consider that this should be RepetitionTime even in pybids, but just noting that there may be cases.

```
RepetitionTime:
  name: Repetition Time
  variable_name: TR
```

---

I kind of liked @effigies' restructuring primarily because it avoided a "variable_name" or "altname" field, while still retaining the same information. I can see metadata fields where a variable name might need to be shorter than a meaningful concept name though.